dbms/src/Storages/KVStore/Read/LearnerRead.cpp

```diff
@@ -356,23 +359,23 @@ LearnerReadSnapshot doLearnerRead(
 
     const size_t num_regions = regions_info.size();
 
-    // TODO: refactor this enormous lambda into smaller parts
+    const auto start_time = Clock::now();
     UnavailableRegions unavailable_regions;
     LearnerReadStatistics stats;
-    const auto batch_wait_index = [&](const size_t region_begin_idx) -> void {
+
+    // TODO: refactor this enormous lambda into smaller parts
```

Comment:
the `batch_wait_index` is removed, so this is not a todo any more?

Reply:
The code block of read index && wait index still takes about 150 lines, I think we could break it into smaller parts (and add some unit testing) later